geomstats/geometry/hpd_matrices.py

```diff
@@ -0,0 +1,1354 @@
+    power_affine : int
+        Power transformation of the classical HPD metric.
+        Optional, default: 1.
+    scale : int or float
+        Scale of the HPD matrices metric.
+        Optional, default: 1.
+    """
+
+    def __init__(self, n, power_affine=1, scale=1):
+        dim = int(n * (n + 1) / 2)
+        super().__init__(
+            dim=dim,
+            shape=(n, n),
+            signature=(dim, 0),
+        )
+        self.n = n
+        self.power_affine = power_affine
+        self.scale = scale
+
+    @staticmethod
+    def _aux_inner_product(tangent_vec_a, tangent_vec_b, inv_base_point):
+        """Compute the inner-product (auxiliary).
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., n, n]
+        tangent_vec_b : array-like, shape=[..., n, n]
+        inv_base_point : array-like, shape=[..., n, n]
+
+        Returns
+        -------
+        inner_product : array-like, shape=[...]
+        """
+        aux_a = ComplexMatrices.mul(inv_base_point, tangent_vec_a)
+        aux_b = ComplexMatrices.mul(inv_base_point, tangent_vec_b)
+
+        # Use product instead of matrix product and trace to save time
+        inner_product = ComplexMatrices.trace_product(aux_a, aux_b)
+
+        return inner_product
+
+    def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
+        """Compute the affine-invariant inner-product.
+
+        Compute the inner-product of tangent_vec_a and tangent_vec_b
+        at point base_point using the affine invariant Riemannian metric.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        tangent_vec_b : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., n, n]
+            Base point.
+
+        Returns
+        -------
+        inner_product : array-like, shape=[..., n, n]
+            Inner-product.
+        """
+        power_affine = self.power_affine
+        hpd_space = HPDMatrices
+
+        if power_affine == 1:
+            inv_base_point = GeneralLinear.inverse(base_point)
+            inner_product = self._aux_inner_product(
+                tangent_vec_a, tangent_vec_b, inv_base_point
+            )
+        else:
+            modified_tangent_vec_a = hpd_space.differential_power(
+                power_affine, tangent_vec_a, base_point
+            )
+            modified_tangent_vec_b = hpd_space.differential_power(
+                power_affine, tangent_vec_b, base_point
+            )
+            power_inv_base_point = HermitianMatrices.powerm(base_point, -power_affine)
+            inner_product = self._aux_inner_product(
+                modified_tangent_vec_a, modified_tangent_vec_b, power_inv_base_point
+            )
+
+            inner_product = inner_product / (power_affine**2)
+
+        inner_product *= self.scale**2
```

Comment:
why is this different from the corresponding `SPDAffineMetric`?

Reply:
The function `inner_product` calls to class method `differential_power`.
In `HPDMatrices.differential_power`, we add a casting to complex which is not in `SPDMatrices.differential_power`.
The class method `differential_power` calls the static method `aux_differential_power`.
In `HPDMatrices.aux_differential_power`, we use a `ComplexMatrices.transconjugate` operator instead of the `Matrices.transpose` operator present in `SPDMatrices.aux_differential_power`.